### PR TITLE
Makefile targets, lint rc file, and dev requirements. Fix formatting.

### PR DESCRIPTION
**Pull request by jspacek**
## Description
- Add a Makefile for test, format, lint, and dev dependencies targets. 
- Linter is recommended one from https://google.github.io/styleguide/pylintrc
- Auto-formatting is recommended `yapf` from https://google.github.io/styleguide/pyguide.html using parameters `style='{based_on_style: google, indent_width: 4}'`
- Add `yapf` to poetry dependencies
- This code is for issue #19.

## Affected Dependencies
`pytest` `pylint` `yapf`

## How has this been tested?
- Tested with `conda` environment under `Python 3.8`

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [n/a] My changes are covered by tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+.PHONY: format
+format:
+	yapf -i --style='{based_on_style: google, indent_width: 2}' -r pipeline_dp tests
```
Comment (chinmayshah99) [inline Makefile:5]: can you add `yapf` to poetry dev too?

**Review by chinmayshah99 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (jspacek) [reply Makefile:5]: 👀 Missed that, thank you!

Comment (jspacek) [reply Makefile:5]: 👋 I ran `poetry add yapf` which added the line `yapf = "^0.31.0"`.
Please let me know if there's another step I should run. Thanks!

Comment (dvadym) [inline Makefile:5]: According to Google style guide indent is 4 spaces
https://google.github.io/styleguide/pyguide.html#34-indentation

**Review by dvadym — COMMENTED** [review]
That's cool, thanks! 

1 minor comment

File: Makefile (hunk 1 above)
Comment (jspacek) [reply Makefile]: Apologies! I thought it was changed to `2`. I updated the Makefile and ran `yapf` for 4 spaces 👍

**Review by dvadym — APPROVED** [review]
Thanks!

**Pull request merged**